doc/porting_guide.md

```diff
@@ -3,268 +3,292 @@
-
     - expose_fs         - adds a new Xively C Client API function which allows external definition
-                          of File System calls
+                          of File System calls.  This will be part of our BSP system in the future.
 
 ###### Development flags
 
-    - memory_limiter    - turns on memory limiting and monitoring functionality. The purpose of this configuration
-                          option is to aid development processes by simulating low-memory environment behavior.
+    - memory_limiter    - turns on memory limiting and monitoring functionality. The purpose of this configuration is to aid development processes by simulating low-memory environment behavior.
                           The memory monitor part is useful for memory consumption tracking and to hunt down memory
-                          leaks.
+                          leaks.  When a leak occurs a stack trace of the initial allocation will be logged.
     - mqtt_localhost    - the Client will connect to localhost MQTT server. The purpose of this configuration option
                           is to help development processes.
     - no_certverify     - lowers security by disabling TLS certificate verification
-    - tls               - build with TLS module for securely connecting with TLS 1.2 to servers. The lack of this
-                          flag will skip the TLS handshaking which may helpful in connecting to mosquitto to test,
+    - tls               - the Client will use third party TLS 1.2 implementations to encrypt data before it's sent over network sockets.   The lack of this
+                          flag will skip the TLS handshaking.  This be may helpful in connecting to mosquitto to test,
                           but may not be supported on some secure services and is not recommend for production
-                          deployment
+                          deployment.  The Xively Gateway will not accept connections without TLS.
 
 ###### Platform selector flags
 
     - posix_platform    - selects implementation for non-BSP time and non-BSP memory solutions
     - wmsdk_platform    - selects implementation for non-BSP time and non-BSP memory solutions
 
-#### Example make command
+The Platform Selector configurations will eventually be sunset.  Currently these configure the build system to include Critical Section implementations for invoking callbacks on new threads.  
+
+We suggest defining `posix_platform` and omit `threading` from your CONFIG options when building for custom platforms.  `threading` is currently ommitted by default.
+
+For more information about thread safe callback support please see the Xively C Client User Guide: `doc/user_guide.md`.
+
+#### Example make Command
 
 By executing a simple 'make' under directory xi\_client\_c should be sufficient on OSX or Linux/Unix. This will result in build configuration with the following default flags:
```

Comment:
as I saw this is the last occurence of xi_client_c.
Maybe "... under root directory should be sufficient..." is ok, isn't it?